### PR TITLE
Added notification for a tutor that has pending students, added way t…

### PR DESCRIPTION
**Pull request by NickIs211**
…o view all pending students, added way to accept pending students

### CODE REVIEW

**Comment by NickIs211** [comment]
Also removed .history folder and added to .gitignore

File: .rubocop.yml [hunk 1]
```diff
@@ -39,3 +39,5 @@ RSpec/MultipleExpectations:
 RSpec/ExampleLength:
   Exclude:
     - 'spec/feature/*'
+Layout/EndOfLine:
+   Enabled: False
```
Comment (baileyjrob) [inline .rubocop.yml]: This shouldn't need changing. I'm hesitant to change things from the baseline as is, and this is an easy-enough thing that I don't see the purpose in changing it.
```suggestion
```

File: app/models/tutoring_session.rb [hunk 2]
```diff
@@ -10,7 +10,6 @@ class TutoringSession < ApplicationRecord
   has_many :departments, through: :department_tutoring_sessions
   # has_and_belongs_to_many :users
   has_many :tutoring_session_users, dependent: :delete_all
-  has_many :users, through: :tutoring_session_users
```
Comment (baileyjrob) [inline app/models/tutoring_session.rb]: I'm pretty sure this line is necessary, I don't understand why it was removed.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -23,13 +23,6 @@
 </head>
 
 <body>
-
-
-
-  <div devise notices and allerts>
-    <p class="notice"><%= notice %></p>
-    <p class="alert"><%= alert %></p>
-  </div>
```
Comment (baileyjrob) [inline app/views/layouts/application.html.erb:32]: Why were these removed? Do we not need devise alerts/notices anymore? I know you've implemented a notification system, but is devise properly routed through the custom notification model?

File: app/views/tutoring_session/show.html.erb [hunk 4]
```diff
@@ -14,8 +14,8 @@
         Current Users:
       </p>
       <p>
-      <% @tsession.users.each do |user| %>
-        <%= user.first_name %> <%= user.last_name %> 
+      <% @tsession.tutoring_session_users.each do |link| %>
+        <%= link.user.first_name %> <%= link.user.last_name %> | <%= link.link_status %>
```
Comment (baileyjrob) [inline app/views/tutoring_session/show.html.erb:18]: What's the purpose of this change? Does anything benefit by using the join-table?

File: app/views/tutoring_session_user/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      //         .draw();
+      //     });
+
+      //   column.data().unique().sort().each(function (d, j) {
+      //     select.append('<option value="'+d+'">'+d+'</option>')
+      //   });
+      // });
+    });
+  </script>
+</div>
```
Comment (baileyjrob) [inline app/views/tutoring_session_user/show.html.erb]: Missing newline
```suggestion
</div>

```

File: db/migrate/20210330003039_add_index_to_tutoring_session_users.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToTutoringSessionUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :tutoring_session_users, :id, :primary_key
```
Comment (baileyjrob) [inline db/migrate/20210330003039_add_index_to_tutoring_session_users.rb:3]: Join tables are not supposed to have ids

**Review by baileyjrob — CHANGES_REQUESTED** [review]
I'm a bit confused about some changed, and a little worried about others. Do we need to work directly with the TutoringSessionUsers table in order to implement all of these changes? That would increase complexity quite a bit.

File: .rubocop.yml (hunk 1 above)
Comment (NickIs211) [reply .rubocop.yml]: That only gives me errors on local, every time I push GitHub changes it to the proper endings so it never is an issue in the dev branch. You can ignore this change if you want.

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (NickIs211) [reply app/views/layouts/application.html.erb:32]: That wasn't removed, it was moved to a bit below in the same file

File: app/models/tutoring_session.rb (hunk 2 above)
Comment (NickIs211) [reply app/models/tutoring_session.rb]: Might have been when I was working on the link table, ignore the change. The helper method can likewise be removed

File: app/views/tutoring_session/show.html.erb (hunk 4 above)
Comment (NickIs211) [reply app/views/tutoring_session/show.html.erb:18]: Link status is only viewable through the join table

File: db/migrate/20210330003039_add_index_to_tutoring_session_users.rb (hunk 6 above)
Comment (NickIs211) [reply db/migrate/20210330003039_add_index_to_tutoring_session_users.rb:3]: An issue was occuring where the join table was being changed, and then rails yelled that it couldn't find the join table. I attempted to have it pk'd via [user_id, tutoring_session_id] however it was not working. In an ideal world, we would index off those two columns but rails sucks and it'd take too much effort compared to just adding this pk.

Comment (baileyjrob) [reply db/migrate/20210330003039_add_index_to_tutoring_session_users.rb:3]: Wait, did you attempt to set user_id and tutoring_session_id to primary keys, or did you index them? Because those are two different things.

Comment (NickIs211) [reply db/migrate/20210330003039_add_index_to_tutoring_session_users.rb:3]: Both at some point, but I attempted in different ways to set primary_key specifically

**Review by baileyjrob — APPROVED** [review]
Everything cleared up
